AutoNode/validator.py

```diff
@@ -0,0 +1,170 @@
+    curr_headers = get_latest_headers("http://localhost:9500/")
+    curr_epoch_shard = curr_headers['shard-chain-header']['epoch']
+    curr_epoch_beacon = curr_headers['beacon-chain-header']['epoch']
+    ref_epoch = get_latest_header(node_config['endpoint'])['epoch']
+    has_looped = False
+    while curr_epoch_shard != ref_epoch or curr_epoch_beacon != ref_epoch:
+        sys.stdout.write(f"\rWaiting for node to sync: shard epoch ({curr_epoch_shard}/{ref_epoch}) "
+                         f"& beacon epoch ({curr_epoch_beacon}/{ref_epoch})")
+        sys.stdout.flush()
+        has_looped = True
+        time.sleep(2)
+        assert_no_bad_blocks()
+        curr_headers = get_latest_headers("http://localhost:9500/")
+        curr_epoch_shard = curr_headers['shard-chain-header']['epoch']
+        curr_epoch_beacon = curr_headers['beacon-chain-header']['epoch']
+        wait_for_node_response(node_config['endpoint'], verbose=False)
+        ref_epoch = get_latest_header(node_config['endpoint'])['epoch']
+    if has_looped:
+        print("")
+    print(f"{Typgpy.OKGREEN}Node synced to current epoch{Typgpy.ENDC}")
+
+
+def _create_new_validator():
+    wait_for_node_response(node_config['endpoint'], verbose=True)
+    print(f"{Typgpy.HEADER}Checking validator...{Typgpy.ENDC}")
+    staking_epoch = get_staking_epoch(node_config['endpoint'])
+    curr_epoch = get_current_epoch(node_config['endpoint'])
+    print(f"{Typgpy.OKBLUE}Verifying Epoch...{Typgpy.ENDC}")
+    while curr_epoch < staking_epoch:  # WARNING: using staking epoch for extra security of configs.
+        sys.stdout.write(f"\rWaiting for staking epoch ({staking_epoch}) -- current epoch: {curr_epoch}")
+        sys.stdout.flush()
+        time.sleep(8)  # Assumption of 8 second block time...
+        assert_no_bad_blocks()
+        wait_for_node_response(node_config['endpoint'], verbose=True)
+        curr_epoch = get_current_epoch(node_config['endpoint'])
+    print(f"{Typgpy.OKGREEN}Network is at or past staking epoch{Typgpy.ENDC}")
+    print(f"{Typgpy.OKBLUE}Verifying Balance...{Typgpy.ENDC}")
+    # Check validator amount +1 for gas fees.
+    if not check_min_bal_on_s0(validator_config['validator-addr'], validator_config['amount'] + 1, node_config['endpoint']):
+        print(f"{Typgpy.FAIL}Cannot create validator, {validator_config['validator-addr']} "
+              f"does not have sufficient funds.{Typgpy.ENDC}")
+        raise RuntimeError("Create Validator Error")
+    else:
+        print(f"{Typgpy.OKGREEN}Address: {validator_config['validator-addr']} has enough funds{Typgpy.ENDC}")
+    _verify_node_sync()
+    print(f"{Typgpy.OKBLUE}Sending create validator transaction...{Typgpy.ENDC}")
+    _send_create_validator_tx()
+
+
+def _send_create_validator_tx():
+    wait_for_node_response(node_config['endpoint'], verbose=True)
+    try:
+        response = cli.single_call(f'hmy --node={node_config["endpoint"]} staking create-validator '
```

Comment:
why doesn't single_call just raise the exception? you use exceptions a lot but the point of exception is to control flow. This isn't control flow, its like giving more info on error, in which case just make the original exploding thing give the informative error. 

try/except is heavy on syntax in python - try to use it less

Reply:
yes it is not needed, it's removed in latest